extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -80,16 +81,48 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
+        scheduler.execute(this::initializeAsync);
+    }
+    
+    private synchronized void initializeAsync() {
         TradfriGatewayConfig configuration = getConfigAs(TradfriGatewayConfig.class);
-        if (configuration.host == null || configuration.host.isEmpty()) {
+
+        if (isNullOrEmpty(configuration.host)) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                     "Host must be specified in the configuration!");
             return;
         }
-        if (configuration.code == null || configuration.code.isEmpty()) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "Security code must be provided in the configuration!");
-            return;
+        if (isNullOrEmpty(configuration.code)) {
+            if (isNullOrEmpty(configuration.identity) || isNullOrEmpty(configuration.preSharedKey)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Either security code or identity and pre-shared key must be provided in the configuration!");
+                return;
+            }
+        } else {
+            configuration.identity = UUID.randomUUID().toString().replace("-", "");
```

Comment:
You must not change configuration values here. As mentioned before, in order to change configuration settings, you have to use editConfiguration/updateConfiguration methods.

Reply:
Here the ```configuration``` variable is used as temporary storage to store a value. The editConfiguration/updateConfiguration methods will be used later when all required data is retrieved and validated.

Reply:
> Here the configuration variable is used as temporary storage to store a value.

No, you are modifying the instance of the current Thing, which is not permitted.

Reply:
Really ? - I thought that ```getConfigAs(Class<T> configurationClass)``` returns a copy of the thing configuration and modifying the returned object doesn't actually alter the configuration map of the thing.

Reply:
Ok, you are right - this implementation indeed creates a new instance, so there is no risk. Nonetheless, the contract of the method does not guarantee you this and the docs specifically state that you must use edit/update if you want to change a configuration. So following this is definitely the better option in case any refactorings are done within the framework in future.